src/GoTrueClient.ts

```diff
@@ -387,34 +432,40 @@ export default class GoTrueClient {
   }
 
   /**
-   * Returns the user data based on the current session, refreshing the session if necessary.
+   * Gets the current user details if there is an existing session.
    */
-  async getUserFromSession(): Promise<
-    | UserResponse
-    | {
-        data: {
-          user: null
+  async getUser(): Promise<UserResponse> {
+    try {
+      let access_token = ''
+      if (!('Authorization' in this.headers)) {
+        const { session, error } = await this.getSession()
+        if (error) {
+          throw error
         }
-        error: null
+        if (!session) {
+          throw new AuthSessionMissingError()
+        }
+        access_token = session.access_token
+      }
+      return await _request(this.fetch, 'GET', `${this.url}/user`, {
+        headers: this.headers,
+        jwt: access_token,
+        xform: _userResponse,
+      })
+    } catch (error) {
+      if (isAuthError(error)) {
+        return { data: { user: null }, error }
       }
-  > {
-    const { session, error } = await this.getSession()
-    if (error) {
-      return { data: { user: null }, error }
-    }
 
-    if (!session) {
-      return { data: { user: null }, error: null }
+      throw error
     }
-
-    return { data: { user: session.user }, error: null }
   }
 
   /**
    * Updates user data, if there is a logged in user.
    * @param attributes The data you want to update.
    */
-  async update(attributes: UserAttributes): Promise<UserResponse> {
+  async updateUser(attributes: UserAttributes): Promise<UserResponse> {
```

Comment:
@alaister choosing not to ignore the value set in `this.headers` because `updateUser` actually updates the user in gotrue and also updates the session with the updated user details - maybe we should have a separate method for just updating the user so it can be used in a server-side context?

Reply:
Does it update the JWT as well as the user's details? If it does we'll either need a new method or a config param for this one